### PR TITLE
All tests passing for database models and API endpoints

### PR DESCRIPTION
**Pull request by jduell12**
# Description
Incorporates tests for the API endpoints and database model functions to make sure the database is operating how it should. Changes were made to the database schema, previous model functions and previous endpoints to allow for a database that allows dynamic data changes. 

## Type of change
Please delete options that are not relevant.
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update

## Change Status
-  Complete, tested, ready to review and merge

# How Has This Been Tested?
- Jest 

# Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Comment by MaryamMosstoufi** [comment]
🔥

File: api/incidents/incidentsRouter.js [hunk 1]
```diff
@@ -4,8 +4,10 @@ const axios = require('axios');
 
 // Model and util imports
 const Incidents = require('./incidentsModel');
+const Sources = require('../sources/sourcesModel');
+const Tags = require('../tags/tagsModel');
 // const { post } = require('../dsService/dsRouter');
```
Comment (ryan-hamblin) [inline api/incidents/incidentsRouter.js:9]: If it's commented out, we probably don't need in main.

File: api/incidents/incidentsRouter.js [hunk 2]
```diff
@@ -142,12 +134,12 @@ router.get('/showallincidents', async (req, res) => {
...
   req.body.forEach((incident) => {
     Incidents.createIncident(incident)
 
-      .then((post) => {
-        res.status(201).json(post);
+      .then((success) => {
+        res.status(201).json(success);
       })
       .catch((err) => {
         console.log(err);
```
Comment (ryan-hamblin) [inline api/incidents/incidentsRouter.js:145]: Don't need console.logs()

File: api/incidents/incidentsRouter.js [hunk 3]
```diff
@@ -229,15 +221,15 @@ router.get('/sources', (req, res) => {
...
+  Sources.getSourcesByIncidentId(id).then((response) => {
     res.json(response);
   });
 });
 
 /**
  * @swagger
  * /createsource:
- *  get:
+ *  post:
```
Comment (ryan-hamblin) [inline api/incidents/incidentsRouter.js:232]: DOCS!

File: api/incidents/middleware/index.js [hunk 4]
```diff
@@ -2,23 +2,42 @@ const db = require('../../../data/db-config');
...
-      incident.city &&
-      incident.id &&
-      incident.state &&
-      incident.title &&
-      incident.desc &&
-      incident.date
-    ) {
-      const id = db('incidents').where('id', incident.id);
-      if (!id[0]) return incident;
+  if (req.body.length > 0) {
```
Comment (ryan-hamblin) [inline api/incidents/middleware/index.js:9]: Maybe tell me about what we are looking for here and why a 500 is necessary if the req body is empty.

**Review by ryan-hamblin — COMMENTED** [review]
Overall here I'm very impressed with the work that was done. Please respond to the areas that I commented on. I'd love to know more about that conditional and exactly what you're checking for there. 

I'd recommend splitting this work up into 2 different PR's so that they're more focused on a single task, and easier to review.

**Pull request merged**